packages/core/android/src/main/kotlin/com/datadog/reactnative/DatadogSDKWrapper.kt

```diff
@@ -21,6 +26,45 @@ import com.datadog.android.trace.Trace
 import com.datadog.android.trace.TraceConfiguration
 import com.datadog.android.webview.WebViewTracking
 
+/**
+ * Internal object used to add internal testing.
+ */
+object DatadogSDKWrapperStorage {
+    internal val onInitializedListeners: MutableList<(InternalSdkCore?) -> Unit> = mutableListOf()
+
+    /**
+     * Adds a Listener called when the core is initialized.
+     */
+    fun addOnInitializedListener(listener: (InternalSdkCore?) -> Unit) {
+        onInitializedListeners.add(listener)
+    }
+
+    /**
+     * Exposed for testing purposes only.
+     */
+    fun notifyOnInitializedListeners() {
+        onInitializedListeners.forEach {
+            it(core)
+        }
+    }
+
+    private var core: InternalSdkCore? = null
+
+    /**
+     * Sets instance of core SDK to be used to initialize features.
+     */
+    fun setSdkCore(core: InternalSdkCore?) {
+        this.core = core
+    }
+
+    /**
+     * Returns the core used for registering RN features.
+     */
+    fun getSdkCore(): SdkCore {
+        return core ?: Datadog.getInstance()
```

Comment:
Maybe some log entry is needed when fallback to the default one is done? It may help debugging.